src/stan/analyze/mcmc/compute_potential_scale_reduction.hpp

```diff
@@ -0,0 +1,162 @@
+#ifndef STAN_ANALYZE_MCMC_COMPUTE_POTENTIAL_SCALE_REDUCTION_HPP
+#define STAN_ANALYZE_MCMC_COMPUTE_POTENTIAL_SCALE_REDUCTION_HPP
+
+#include <stan/math/prim/mat.hpp>
+#include <stan/analyze/mcmc/autocovariance.hpp>
+#include <stan/analyze/mcmc/split_chains.hpp>
+#include <boost/math/special_functions/fpclassify.hpp>
+#include <algorithm>
+#include <cmath>
+#include <vector>
+#include <limits>
+
+namespace stan {
+namespace analyze {
+
+  /**
+   * Computes the potential scale reduction (Rhat) for the specified
+   * parameter across all kept samples.
+   *
+   * See more details in Stan reference manual section "Potential
+   * Scale Reduction". http://mc-stan.org/users/documentation
+   *
+   * Current implementation assumes chains are all of equal size and
+   * draws are stored in contiguous blocks of memory.
+   *
+   * @param draws stores pointers to arrays of chains
+   * @param sizes stores sizes of chains
+   * @return potential scale reduction for the specified parameter
+   */
+  inline
+  double compute_potential_scale_reduction(std::vector<const double*> draws,
+                                           std::vector<size_t> sizes) {
+    int num_chains = sizes.size();
+    size_t num_draws = sizes[0];
+    for ( int chain = 1; chain < num_chains; ++chain ) {
```

Comment:
The function doc says the code assumes that the chains are all of equal size, but this code doesn't.  Either the function doc should change ("chains are trimmed from the back to match the length of the shortest chain") or this code should check equal size and throw a NaN or an exception if they're not the same.

Reply:
Made explicit the trimming.